### DISCUSSION TITLE
I have three libraries, mixed formats, DECADES old. Help me organize it?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [beetbox/beets#4598] [General]
Hello!

I started to use a PC in the late 90s (born '93) and that quickly included ripping my music to my hard disk. And later during school, we obviously also exchanged music; some came from iPods, some from mp3 players and what not.

Over the years, this has become an absolute mess. iTunes metadata, Windows Media Player Audio formats, inconsistent folder structures due to downloads and other things - and just so much more...

My question is: Can I point Beets at this (and two "offspring") libraries and just have it:

- Deduplicate all the files
- Create a proper hierachy of all of them
- Convert all that isn't in a good format (i.e. WMA) to an actually acceptable one
- Whilst operating on network drives?

Said network drives are SMB mounts on my home server partially via RClone and regular smb.

Oh and this library is about 300GB-ish big... last time i had it scanned.

Thanks and kind regards,
Ingwie

**Comment by sampsyo** [comment]
I definitely think it's worth a shot! It won't exactly be fast, and I would proceed with caution (make a backup first, work on a small subset of music before addressing the whole library). But for very messy libraries, options like `--group-albums` can be effective. I might also suggest importing everything first *without* autotagging (`beet import -A`) to get an initial database built up, and then going through and looking at how chaotic things are. Re-importing to get tags from MusicBrainz is always an option.

**Reply by IngwiePhoenix** [reply]
Awesome, thanks!

I have all my libraries now linked to a NAS - so I can just throw beets in a `screen`and go to sleep :)

**Reply by IngwiePhoenix** [reply]
```
root@FriendlyWrt ~# beet stats
Tracks: 28434
Total time: 10.6 weeks
Approximate total size: 196.1 GiB
Artists: 7420
Albums: 1381
Album artists: 296
```
Yep, using `screen` really was a good idea. xD
Now with all of this in, how can I run the auto tagger in post? I `import -A`'d.

**Reply by RollingStar** [reply]
Try a backup first. To get started:

https://beets.readthedocs.io/en/latest/reference/cli.html?highlight=reimport#reimporting

I vaguely remember this deleting my files somehow.
